src/intrinsic/mod.rs

```diff
@@ -726,51 +726,43 @@ impl<'a, 'gcc, 'tcx> Builder<'a, 'gcc, 'tcx> {
                 "__builtin_clzll"
             }
             else if width == 128 {
-                // Algorithm from: https://stackoverflow.com/a/28433850/389119
-                let array_type = self.context.new_array_type(None, arg_type, 3);
-                let result = self.current_func()
-                    .new_local(None, array_type, "count_loading_zeroes_results");
-
+                // Algorithm from: https://stackoverflow.com/a/28433850/389119 updated to check for high 64bits being 0
```

Comment:
It seems your new algorithm significantly diverges from the one linked here.
If so, could you please provide the link to the algorithm you used? If you can't, could you please add comments below to make this easier to understand?